### PR TITLE
Prevent borrower from repaing with surplus

### PR DESCRIPTION
**Pull request by MiksuJak**

### CODE REVIEW

File: contracts/truefi/LoanToken.sol [hunk 1]
```diff
@@ -301,6 +301,9 @@ contract LoanToken is ILoanToken, ERC20 {
      * @param _amount amount of currencyToken to repay
      */
     function repay(address _sender, uint256 _amount) external override onlyAfterWithdraw {
+        if (_amount > debt) {
```
Comment (vanruch) [inline contracts/truefi/LoanToken.sol:304]: Revert in this case. It's better to throw mistake than quitely fix it and lead to user's confusion.

Comment (barrutko) [reply contracts/truefi/LoanToken.sol:304]: I agree. Especially considering the fact, that this amount will not move unexpectedly and frontend app can fetch it and pass accurately.

**Comment by MiksuJak** [comment]
Requested changes have just come. 
May I also suggest changing a condition in close() function?
`
function close() external override onlyOngoing {
        require(start.add(term) <= block.timestamp, "LoanToken: Loan cannot be closed yet");
        if (_balance() >= debt) {
            status = Status.Settled;
        } else {
            status = Status.Defaulted;
        }

        emit Closed(status, _balance());
    }
`
Considering the fact, that now repaying with surplus is impossible, we can strictly check if contract balance equals the debt.
From my initial research, this might require to rewrite a few tests.

**Comment by vanruch** [comment]
> Requested changes have just come.
> May I also suggest changing a condition in close() function?
> 
> ```
> function close() external override onlyOngoing {
>         require(start.add(term) <= block.timestamp, "LoanToken: Loan cannot be closed yet");
>         if (_balance() >= debt) {
>             status = Status.Settled;
>         } else {
>             status = Status.Defaulted;
>         }
> 
>         emit Closed(status, _balance());
>     }
> ```
> 
> Considering the fact, that now repaying with surplus is impossible, we can strictly check if contract balance equals the debt.
> From my initial research, this might require to rewrite a few tests.

It's still possible, by doing a simple transfer. We cannot prevent this really

**Review by vanruch — APPROVED** [review]

**Pull request merged**